integrationTests/vm/txsFee/scCalls_test.go

```diff
@@ -300,3 +362,47 @@ func TestESDTScCallAndGasChangeShouldWork(t *testing.T) {
 		require.Nil(t, errCommit)
 	}
 }
+
+func TestScCallDistributeStakingRewards_ShouldWork(t *testing.T) {
+	testContext, scAddress := prepareTestContextForEpoch836(t)
+	defer testContext.Close()
+
+	pkConv, _ := pubkeyConverter.NewBech32PubkeyConverter(32, log)
+	sndAddr1, err := pkConv.Decode("erd1rkhyj0ne054upekymjafwas44v2trdykd22vcg27ap8x2hpg5u7q0296ne")
+	require.Nil(t, err)
+
+	senderBalance := big.NewInt(1000000000000000000)
+	gasPrice := uint64(10)
+	gasLimit := uint64(600000000)
+
+	_, _ = vm.CreateAccount(testContext.Accounts, sndAddr1, 0, senderBalance)
+
+	blockChainHook := testContext.BlockchainHook.(process.BlockChainHookHandler)
+
+	utils.CleanAccumulatedIntermediateTransactions(t, testContext)
+	blockChainHook.SetCurrentHeader(&block.Header{
+		TimeStamp: 1668430842,
+	})
+
+	txData, errDecode := hex.DecodeString("646973747269627574655f7374616b696e675f72657761726473")
+	require.Nil(t, errDecode)
+	tx := vm.CreateTransaction(0, big.NewInt(0), sndAddr1, scAddress, gasPrice, gasLimit, txData)
+
+	startTime := time.Now()
+	returnCode, errProcess := testContext.TxProcessor.ProcessTransaction(tx)
+	endTime := time.Now()
+	assert.Nil(t, errProcess)
+	assert.Equal(t, vmcommon.Ok, returnCode)
+
+	_, errCommit := testContext.Accounts.Commit()
+	require.Nil(t, errCommit)
+
+	intermediateTxs := testContext.GetIntermediateTransactions(t)
+	assert.Equal(t, 1, len(intermediateTxs))
+	log.Info(spew.Sdump(intermediateTxs))
```

Comment:
might print them in a prettier way (bech32 addresses, and so on)

Reply:
done